### DISCUSSION TITLE
Gleam for embedded linux projects

### DISCUSSION THREAD

**Discussion by joaohf** [gleam-lang/gleam#3791] [General]
Hi,

I'm wondering if Gleam could be useful for developing embedded linux projects. Think about something near [Nerves](https://nerves-project.org/) (just an example to you get the idea. I don't want comparatives), but written in Gleam.

Nowadays one can develop an embedded solution that runs on Linux and is written in Elixir or Erlang. Why not bring also Gleam as an feasible option too ?

Talking about BEAM specifically, there is a few options for embedded area. To get a better view, I recommended this reading [BEAM on Embedded Landscape 2020](https://erlef.org/wg/embedded/content/landscape).

So, I just want to introduce Gleam as an option. For that, I integrated Gleam to the meta-erlang project. The blog post called [Introducing Gleam to meta-erlang layer](https://meta-erlang.github.io/blog/2024/11/03/index/) explains a bit more the idea.

Well, [meta-erlang](https://meta-erlang.github.io/) is a [Yocto Project](https://www.yoctoproject.org/) layer that brings BEAM ecosystem to the embedded Linux projects.

I've been maintaining meta-erlang as a feasible way to run Erlang and Elixir programs on linux distributions made with Yocto. It is a long story. But now Gleam is an option too.

I'm looking for some prototype ideas (I have many but can't decide). Maybe you have one about a small demo or PoC that we can write in Gleam  just to compare or something else.

Thanks

**Comment by lpil** [comment]
That's great! Very cool.

I believe the Nerves folks are doing something similar. Perhaps they have some cool demo ideas too?
